rpc/openapi/openapi.yaml

```diff
@@ -601,6 +601,31 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/ErrorResponse"
+  /unsafe_flush_mempool:
+    get:
+      summary: Flush mempool of all unconfirmed transactions
+      operationId: unsafe_flush_mempool
+      tags:
+        - Unsafe
+      description: 
+        Flush flushes out the mempool. It acquires a read-lock, fetches all the
```

Comment:
I'm seeing that this [is accurate](https://github.com/tendermint/tendermint/blob/6909158933b5a7b691ced7d279039bf2bc8ed2d5/internal/mempool/v1/mempool.go#L314) but it seems like incorrect behavior since it clears out the mempool cache, an action that seems like it should be guarded by a write lock.

Reply:
Yeah I also did a double take on that. Perhaps @alexanderbez can help answer.

Reply:
I looked at the implementation in a bit more detail, and it appears the main uses of the writer lock are for doing updates and modifying notification state (as a side note: The latter isn't documented; I think that lock's scope may have been extended over time). 

The mempool contents appear to be (mostly?) concurrent collections, so at least in theory it should be safe to edit those while there are other readers, as long as you aren't also expecting to post a notification.  All of this is extremely sus, but it could conceivably work. (The results observed by concurrent readers would be inconsistent and non-linearizable, but we did say it was "unsafe" 👀)

All of this does depend on the updater correctly take and releasing the exclusive lock explicitly, though (😢), so to really be sure we'd probably have to audit the usage.